### DISCUSSION TITLE
Save new user to custom user storage: Only username is available in the overridden "addUser" method?

### DISCUSSION THREAD

**Discussion by javac9** [keycloak/keycloak#16073] [Q&A]
I'm struggling to understand how adding a user to my custom storage is supposed to work.

The method that seems to be the one to use is:
```
@Override
public UserModel addUser(final RealmModel realmModel, final String username) { 

}
```

However, here's only the `username` available. Intuitively one expects that all the attributes of the register form would be available so that the new user can be constructed and persisted.

What I did get in the meantime is that if I would, for any reason, return `null` from this method, my custom storage would be disregarded and other ones (if they exist) would be tried, with the fallback to the default keycloak storage.

Can anyone provide some input, it will be appreciated!

**Comment by javac9** [comment]
This someone else's StackOverflow question nicely explains the problem:
Even though for the `addUser()` method one returns a `UserModel` object, and even though the remaining attributes will be added to one's own user object, there is still the question: where does one calls its own RESTful API for example, to do the actual persisting of the new user?

https://stackoverflow.com/questions/60632049/users-created-from-the-admin-console-only-store-the-username-when-using-the-stor

**Reply by cesdperez** [reply]
I'm facing a similar challenge. Were you able to make this work @javac9? I've found where to update the rest of the attributes, but that introduces other challenges. See my question https://github.com/keycloak/keycloak/discussions/20179
